storage/storage.go

```diff
@@ -2076,6 +2076,21 @@ func (wb *withBackoff) apply(config *retryConfig) {
 	config.backoff = &wb.backoff
 }
 
+// WithMaxRetry sets the maximum number of retries for operations that may fail.
+func WithMaxRetry(maxRetryCount int) RetryOption {
```

Comment:
nit: maybe `WithMaxAttempts`? Retry is implied by the type so I don't think it needs to be in the name

Reply:
Done